public/app/features/dashboard-scene/scene/PanelMenuBehavior.tsx

```diff
@@ -264,22 +265,48 @@ export function panelMenuBehavior(menu: VizPanelMenu, isRepeat = false) {
 
     items.push(getInspectMenuItem(plugin, panel, dashboard));
 
-    // TODO: make sure that this works reliably with the reactive extension registry
-    // (we need to be able to know in advance what extensions should be loaded for this extension point, and make it possible to await for them.)
-    const { extensions } = getPluginLinkExtensions({
-      extensionPointId: PluginExtensionPoints.DashboardPanelMenu,
-      context: createExtensionContext(panel, dashboard),
-      limitPerPlugin: 3,
-    });
-
-    if (extensions.length > 0 && !dashboard.state.isEditing) {
-      items.push({
-        text: 'Extensions',
-        iconClassName: 'plug',
-        type: 'submenu',
-        subMenu: createExtensionSubMenu(extensions),
+    const ExtensionsMenuItem = () => {
+      const { links, isLoading } = usePluginLinks({
+        extensionPointId: PluginExtensionPoints.DashboardPanelMenu,
+        context: createExtensionContext(panel, dashboard),
+        limitPerPlugin: 3,
       });
-    }
+
+      if (dashboard.state.isEditing) {
+        return null;
+      }
+
+      if (isLoading) {
+        renderPanelMenuItems([
+          {
+            text: 'Extensions (loading...)',
+            iconClassName: 'plug',
+          },
+        ]);
+      }
+
+      if (links.length === 0) {
+        return null;
+      }
+
+      return (
+        <>
+          {renderPanelMenuItems([
+            {
+              text: 'Extensions',
+              iconClassName: 'plug',
+              type: 'submenu',
+              subMenu: createExtensionSubMenu(links),
+            },
+          ])}
+        </>
+      );
+    };
+
+    items.push({
+      text: 'Extensions',
+      component: ExtensionsMenuItem,
```

Comment:
I assume this will mean we need that scenes PR as well?

Reply:
Instead of having to define a new component every time menu is opened I think it would be cleaner to have the component be a separate standalone component that takes in panel + dashboard as props. 

It would then be 
```tsx
   items.push({
      text: 'Extensions',
      component: <ExtensionsMenuItem panel={panel} dashboard={dashboard} />
```

maybe component should then be "reactNode", alternatively you have component + props. 

The thing that feels messy is having component and then having to create a new component (That captures the panel + dashboard variables via scoped lamda)

Reply:
Thanks, I think that's a good idea! I'll try it out and get back!